src/schema/comments.ts

```diff
@@ -374,6 +483,73 @@ export const resolvers: IResolvers<any, Context> = {
         },
       );
     },
+    recommendedMentions: async (
+      _,
+      { postId, name, limit = 5 }: GQLMentionUserArgs,
+      ctx,
+    ): Promise<GQLMentionUser[]> => {
+      if (name) {
+        const recent = await getRecentMentions(ctx.con, ctx.userId, {
+          name,
+        });
+        const missing = limit - recent.length;
+
+        if (missing === 0) {
+          return recent;
+        }
+
+        const foundUsernames = recent.map(({ username }) => username);
+        const users = await ctx
+          .getRepository(User)
+          .createQueryBuilder()
+          .select('name, username, image')
+          .where('(name ILIKE :name OR username ILIKE :name)', {
+            name: `${name}%`,
+          })
+          .andWhere({
+            username: Not(In(foundUsernames)),
+          })
+          .limit(missing)
+          .getRawMany<GQLMentionUser>();
+
+        return recent.concat(users);
+      }
+
+      const postRepo = ctx.getRepository(Post);
+      const commentRepo = ctx.getRepository(Comment);
+      const postQuery = postRepo
+        .createQueryBuilder('p')
+        .select('u.name, u.username, u.image')
+        .innerJoin(User, 'u', 'u.id = p."authorId"')
+        .where('p.id = :postId', { postId });
+      const commentQuery = commentRepo
+        .createQueryBuilder('c')
+        .select('u.name, u.username, u.image')
+        .innerJoin(User, 'u', 'u.id = c."userId"')
+        .where('c."postId" = :postId AND u.id != :userId', {
+          postId,
+          userId: ctx.userId,
+        })
+        .limit(4);
+
+      const [author, commenters] = await Promise.all([
+        postQuery.getRawMany<GQLMentionUser>(),
+        commentQuery.getRawMany<GQLMentionUser>(),
+      ]);
+
+      const recommendations = [...author, ...commenters];
+      const missing = limit - recommendations.length;
+
+      if (missing === 0) {
+        return recommendations;
+      }
+
+      const recent = await getRecentMentions(ctx.con, ctx.userId, {
+        limit: missing,
+      });
+
+      return recommendations.concat(recent);
```

Comment:
This needs more abstractions. Quite challenging to read. Especially because of all the inline queries